components/BattleScreen/BattleScreen.js

```diff
@@ -1,11 +1,66 @@
-import React, { useState } from "react";
 import styled from "styled-components";
+
+import React, { useState } from "react";
 import PlayerPokemon from "./Pokemon/PlayerPokemon";
 import PlayerState from "./PlayerState/PlayerState";
 import EnemyPokemon from "./Pokemon/EnemyPokemon";
 import EnemyState from "./EnemyState/EnemyState";
 import Menu from "./Menu/Menu";
-import VictoryMessage from "./VictoryMessage/VictoryMessage";
+import VictoryMessage from "./Message/VictoryMessage";
+import LoserMessage from "./Message/LoserMessage";
+
+export default function BattleScreen() {
+	const [enemyHealth, setEnemyHealth] = useState(120);
+	const [playerHealth, setPlayerHealth] = useState(120);
+	const [victory, setVictory] = useState(false);
+	const [isDisabled, setIsDisabled] = useState(false);
+	const [isEnemyDefeated, setIsEnemyDefeated] = useState(false);
+
+	const handleAttack = () => {
+		// Ich greife an
+		const damageDealt = Math.floor(Math.random() * (50 - 1 + 1) + 1);
+		setEnemyHealth(enemyHealth - damageDealt);
+
+		if (enemyHealth - damageDealt <= 0) {
+			setIsEnemyDefeated(true);
+			setVictory(true);
+			return;
+		}
+
+		// Gegner greift zurück an
+		setIsDisabled(true); // Button wird deaktiviert
+		setTimeout(() => {
+			if (isEnemyDefeated) {
+				setIsDisabled(false); // Button wird wieder aktiviert
+				return;
+			}
+
+			const damageTaken = Math.floor(Math.random() * (50 - 1 + 1) + 1);
+			setPlayerHealth(playerHealth - damageTaken);
+			setIsDisabled(false); // Button wird wieder aktiviert
+
+			if (playerHealth - damageTaken <= 0) {
+				setVictory(false);
+			}
+		}, 1000);
```

Comment:
the timeout is quite short, you should do some error handling here, if the jsx component is getting dismounted for example while the timeout is not being cleared, it will crash your app

Reply:
* make sure timeout is being cleared with clearTimeout(<insert timeout>)

Reply:
I have not yet thought of this case, thank you! :-)